File: Ugulamalar/0Yedekler/Excelent - Kopya/Excelent/ThisAddIn.vb
Public Class ThisAddIn

    Private Sub ThisAddIn_Startup() Handles Me.Startup
        '        RegisterComserver()

    End Sub

    Private Sub ThisAddIn_Shutdown() Handles Me.Shutdown

    End Sub
    Private Sub RegisterComserver()
        Try
            Dim strAddinName As String = "Excelent.Fonksiyonlar"

            app.AddIns.Add(Filename:=strAddinName)
            app.AddIns(strAddinName).Installed = True

        Catch ex As Exception
            System.Windows.Forms.MessageBox.Show(ex.ToString)
        End Try
    End Sub
    
End Class

File: Ugulamalar/VSTOcsharp/ThisAddIn.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Excel = Microsoft.Office.Interop.Excel;
using Office = Microsoft.Office.Core;
using System.Windows.Forms;

namespace VSTOcsharp
{
    public partial class ThisAddIn
    {
        private MyUserControl myUserControl1;
        private Microsoft.Office.Tools.CustomTaskPane myCustomTaskPane;

        private void ThisAddIn_Startup(object sender, System.EventArgs e)
        {
            myUserControl1 = new MyUserControl();
            myCustomTaskPane = this.CustomTaskPanes.Add(myUserControl1, "My Task Pane");
            myCustomTaskPane.Visible = true;
            //MessageBox.Show("Merhaba Excel!");
            //Form1 frm1 = new Form1();
            //frm1.Show();
            //Form1 frm2 = new Form1();
            //frm2.Show();
            //Form1 frm3 = new Form1();
            //frm3.Show();        
            if (DateTime.Today>Convert.ToDateTime("31.12.2019"))
            {
                MessageBox.Show("Add'in in süresi dolmuştur.");
                Globals.Ribbons.Ribbon1.Dispose();
            }
            
        }

        private void ThisAddIn_Shutdown(object sender, System.EventArgs e)
        {
        }

        #region VSTO generated code

        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InternalStartup()
        {
            this.Startup += new System.EventHandler(ThisAddIn_Startup);
            this.Shutdown += new System.EventHandler(ThisAddIn_Shutdown);
        }
        
        #endregion
    }
}
